src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -63,27 +68,42 @@ export default App.extend({
     'change': 'onChangeState',
   },
   onChangeState(state) {
+    store.set(this.stateId, state);
     if (state.hasChanged('groupId')) {
       this.restart();
     }
   },
   viewEvents: {
     'toggle:listType': 'onToggleListType',
   },
-  onBeforeStart({ worklistId, worklistType }) {
+  initListState(options) {
+    if (!this.stateId) {
+      this.stateId = `worklist-${ options.worklistId }-${ options.worklistType }`;
+    }
+
+    if (!store.get(this.stateId)) {
+      store.set(this.stateId, DEFAULT_STATE);
+      this.setState(DEFAULT_STATE);
+    } else {
+      this.setState(store.get(this.stateId));
+    }
+  },
+  onBeforeStart(options) {
     if (this.isRestarting()) {
       this.getRegion('list').startPreloader();
       return;
     }
-    this.worklistId = worklistId;
-    this.worklistType = worklistType;
+
```

Comment:
I think what was here was fine without options unless there's a reason `initListState` shouldn't use `this.worklistId`?

Reply:
Nope, no reason.